Sources/Spezi/Spezi.docc/Module/Module Communication.md

```diff
@@ -0,0 +1,55 @@
+# Module Communication
+
+Establish data flow between `Module`s without establishing a dependency hierarchy.  
+
+<!--
+
+This source file is part of the Stanford Spezi open-source project
+
+SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+
+SPDX-License-Identifier: MIT
+
+-->
+
+## Overview
+
+### Communication
+
+``Module``s can easily communicate with each other using the ``Module/Provide`` and ``Module/Collect`` property wrappers.
+
+Upon configuration, the value of each ``Module/Collect`` property will be collected. Therefore,
+all properties must have been property initialized after the initializer of the ``Module`` has been called.
+Before the invocation of ``Module/configure()-5pa83``, the data of all ``Module/Provide`` properties will be made available.
+Refer to the documentation of the property wrappers for a more detailed overview of the available capabilities.
+
+> Important: Accessing `@Provide` properties within the ``Module/configure()-5pa83`` method or accessing `@Collect` properties before
+``Module/configure()-5pa83``, will result in a runtime error. 
+
+Below is a simple example of passing data between ``Module``s.
+
+```swift
+class ModuleA: Module {
+    @Provide var someGreeting = "Hello"
+}
+
+class ModuleB: Module {
+    @Provide var someGreeting = ["Hola", "Hallo"]
+}
```

Comment:
Might be cool to see the initializer being used to set the provide elements to provide an other example?